### ISSUE TITLE
Focal loss migration

### ISSUE THREAD

**Issue by AakashKumarNain** [tensorflow/addons#2004]
@tanzhenyu moving the discussion regarding focal loss from #914  to here. Here is a working colab for the cases you asked.

https://colab.research.google.com/drive/1JIaUVOHC0rTsBWKK-oTBv5qZJ5zLJzTQ?usp=sharing

**Comment by bhack** [comment]
Focal loss now is at https://github.com/tensorflow/models/blob/master/official/vision/keras_cv/losses/focal_loss.py

**Comment by tanzhenyu** [comment]
Thanks @AakashKumarNain . I don't have access to the colab. We do provided a keras_cv focal loss. Let us know if the implementation serves your need.

**Comment by AakashKumarNain** [comment]
Thanks @tanzhenyu . The implementation is good but I have one concern. This loss should be included in core keras because:
1. It is not limited to object detection or computer vision only. Anything that involves cross-entropy loss can leverage focal loss as well. Including it in a `cv` specific package sends a message that it works only for CV which isn't true.
2. Focal loss was introduced in 2017. It has enough citations to be included in the core.

Let me know what you think

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
